app/human-detector-app/services/BackendService.ts

```diff
@@ -0,0 +1,123 @@
+import axios from 'axios';
+import { executeNativeBackPress } from 'react-native-screens';
+import { z } from 'zod';
+import Group from '../classes/Group';
+import User from '../classes/User';
+import Notification from '../classes/Notification';
+import * as ServerUrl from '../config/ServerConfig';
+
+/**
+ * This file handles HTTP requests to our backend
+ */
+
+/**
+ * This method is used to get the group list that is connected to a user.  This group list
+ * will contain cameras and other information that will be stored as a group object.
+ *
+ * @param userId : the userId of the user that we need the group list for.
+ * @returns null if error occurs.
+ *          An array of Groups that are owned by the user with the userId.
+ */
+export async function getGroupListAPI(userId: string): Promise<Group[] | null> {
+  const apiLinkWithExtension: string =
+    ServerUrl.apiLink + ServerUrl.getGroupsListUrlExtension(userId);
+  let testData: any;
+  let success: boolean = false;
+  const config = {
+    headers: {
+      Accept: 'application/json',
+    },
+  };
+
+  await axios
+    .get(apiLinkWithExtension, config)
+    .then((response) => {
+      testData = response.data;
```

Comment:
Axios automatically converts response data to JSON, but it doesn't ensure that this data is well-formed (hence the `any` type on `testData`). We should use Zod to verify that the data we get back is what we expect, but casting is fine for now.

Reply:
ill put it on the backlog so i don't forget lmao